core/src/lib.rs

```diff
@@ -55,25 +50,78 @@ mod peer_record_proto {
 pub use multiaddr;
 pub type Negotiated<T> = multistream_select::Negotiated<T>;
 
-mod peer_id;
+#[deprecated(since = "0.39.0", note = "Depend on `libp2p-identity` instead.")]
+pub mod identity {
+    pub use libp2p_identity::Keypair;
+    pub use libp2p_identity::PublicKey;
```

Comment:
@obi1kenobi I think we might have hit another bug here! `cargo semver-checks` is reporting these two as missing. I've tried with type aliases as well and it doesn't make a difference which makes me think that I am doing something wrong here? It did work in my testing though, i.e. I got heaps of deprecation warnings but stuff was still working when I developed this feature.

Reply:
This is a move to another crate + re-export, right? Semver-checking right now only generates the tested crate's rustdoc JSON, so it doesn't see cross-crate re-exports: https://github.com/obi1kenobi/cargo-semver-checks/issues/355

It's fixable issue with a clear path forward. Just a matter of engineering cycles, which are always in short supply.

Thankfully, once you publish the next version of the affected crate, the false-positive will go away since the baseline will shift to that new version.